src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4721,102 +4810,124 @@ void PhaseIdealLoop::verify_compare( Node *n, const PhaseIdealLoop *loop_verify,
         tty->cr();
       }
     }
+    */
+    return false; // pass
   }
+}
 
-  // Check for immediate dominators being equal
-  if( i >= _idom_size ) {
-    if( !n->is_CFG() ) return;
-    tty->print("CFG Node with no idom: ");
-    n->dump();
-    return;
-  }
-  if( !n->is_CFG() ) return;
-  if( n == C->root() ) return; // No IDOM here
-
-  assert(n->_idx == i, "sanity");
-  Node *id = idom_no_update(n);
-  if( id != loop_verify->idom_no_update(n) ) {
-    tty->print("Unequals idoms for: ");
-    n->dump();
-    if( fail++ > 10 ) return;
-    tty->print("We have it as: ");
-    id->dump();
-    tty->print("Verify thinks: ");
-    loop_verify->idom_no_update(n)->dump();
-    tty->cr();
+void IdealLoopTree::collect_children(GrowableArray<IdealLoopTree*> &children) const {
+  children.clear();
+  IdealLoopTree* child = _child;
+  while (child != nullptr) {
+    assert(child->_parent == this, "all must be children of this");
+    children.push(child);
```

Comment:
You could directly use:
```
children.insert_sorted<compare_tree>(child);
```
instead of `push` + `sort()` afterwards. This would simplify `compare_tree()` to:
```
int compare_tree(IdealLoopTree* const& a, IdealLoopTree* const& b) {
  assert(a != nullptr && b != nullptr, "must be");
  return a->_head->_idx - b->_head->_idx;
}
```